File: Regression/Section 9 - Random Forest Regression/randomForestRegression.py
# Random Forest Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataSet = pd.read_csv('Position_Salaries.csv')
X = dataSet.iloc[:, 1:2].values
y = dataSet.iloc[:, 2].values



# Fitting Random Forest Regression to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 300, random_state = 0)
regressor.fit(X, y)

# Predicting a new result
yPredict = regressor.predict(6.5)

# Visualising the Random Forest Regression results (higher resolution)
xGrid = np.arange(min(X), max(X), 0.01)
xGrid = xGrid.reshape((len(xGrid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(xGrid, regressor.predict(xGrid), color = 'blue')
plt.title('Truth or Bluff (Random Forest Regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

File: Regression/Section 9 - Random Forest Regression/randomForestRegression.R
# Random Forest Regression

# Importing the dataSet
dataSet = read.csv('Position_Salaries.csv')
dataSet = dataSet[2:3]


# Fitting Random Forest Regression to the dataSet
#install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataSet[-2],
                         y = dataSet$Salary,
                         ntree = 500)

# Predicting a new result with Random Forest Regression
yPred = predict(regressor, data.frame(Level = 6.5))

# Visualising the Random Forest Regression results (higher resolution)
# install.packages('ggplot2')
library(ggplot2)
xGrid = seq(min(dataSet$Level), max(dataSet$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataSet$Level, y = dataSet$Salary),
             colour = 'red') +
  geom_line(aes(x = xGrid, y = predict(regressor, newdata = data.frame(Level = xGrid))),
            colour = 'blue') +
  ggtitle('Truth or Bluff (Random Forest Regression)') +
  xlab('Level') +
  ylab('Salary')